draft-ietf-rats-corim.md

```diff
@@ -809,7 +815,8 @@ The following describes each member of the `measurement-map`:
 
 ###### Measurement Keys {#sec-comid-mkey}
 
-The types defined for a measurement identifier are OID, UUID or uint.
+A measurement-associated (sub-)environment is at the granularity of a single element, such as the second enumerated network interface card.
```

Comment:
I would like to, but given the existing use of "measured-element-type-choice" to refer to those logically composed objects, I'd like to know what word we want to use that's not the "element" from the CDDL spec. Do we want to say measured-object?

Reply:
I suppose everyone has different intuition around what words to use to describe compositions of things. Using RFC9334 as a guide, it uses environment (TE/AE) and "sub-Attester" which is another way of saying some sort of construction of TEs and AEs, but it didn't use "sub-TE" or "sub-AE" and some would find this awkward. 

The current wording reflects the idea that a TE can be broken into finer granularity sub-parts and the English dictionary definition of "element" is a word that isn't too overloaded.
If a lay person described a computing environment as something that consisted of several sub-elements it wouldn't be too far from reality while still being abstract enough to fit a wide variety of implementations.

Reply:
I say (sub-)enviroment consistently with the existing text for `mkey`. Notion of it being a sub-environment is that it is further limiting the scope of the subject named by the `environment-map`, but is not itself an element of `environment-map`.

I changed from "element" to "object-id" since I read that you wanted element to consistently mean the same thing, i.e., the meaning from CDDL. Are you suggesting that "element" is not too overloaded as an English term, so we _don't_ have to use "element" to consistently refer to the CDDL concept? I'm confused what you want.

Reply:
>I'm confused what you want.

I'm suggesting that before making changes to CDDL and spec text as it pertains to `mkey`, we should resolve the meta conversation of whether mkey is going to be removed. 
Issue #70 was tracking a use case that possibly justifies its use. The other use cases (that I'm aware of) are profile specific. 

There was a thread related to `mkey` as part of the internal representation conversation that resulted in the following CDDL being included:
```
ECT = {
  ? e: environment-map
  ? c: claims-map / [ + local-claim ]
  ? a: [ + $crypto-key-type-choice ]
  ? ns: text
  ? cm: cm-type
  ? p: $profile-type-choice
}
local-claim = {
  le: local-environment
  c: claims-map
}
local-environment =  bstr / tstr
```
The strongest reason to remove `mkey` is its another form of an ECT structure (that is awkwardly nested in the Claim part of a "parent?" ECT). Since we clarified that the identifier scope for environment values in `environment-map` is global. If an `mkey` has global scope, another ECT could be used instead.  

So, the salient benefit that `mkey` brings is some syntactic sugar when `mkey` identifiers are locally scoped (relative to the parent ECT) to make RIMs less verbose. Instead of having to define another OID or UUID to disambiguate multiple measurements of the same kind (i.e., multiple entries with the same measurement-values-map code point). A local scope identifier, such as a number or string, is used.

But we need to have this conversation in the context of a use case (e.g., issue #70). So, lets prioritize #70 above #248.